gpflow/experimental/check_shapes/__init__.py

```diff
@@ -48,247 +42,282 @@ def linear_model(
 function, or the special name ``return``. The value ``return`` refers to the value returned by the
 function.
 
-The ``<argument specifier>`` can then be modified to refer to elements of the object in two ways:
+The ``<argument specifier>`` can then be modified to refer to elements of the object in several
+ways:
+
+* Use ``.<name>`` to refer to an attribute of an object:
+
+  .. literalinclude:: /examples/test_check_shapes_examples.py
+     :start-after: [argument_ref_attribute]
+     :end-before: [argument_ref_attribute]
+     :dedent:
+
+* Use ``[<index>]`` to refer to a specific element of a sequence. This is particularly useful if
+  your function returns a tuple of values:
+
+  .. literalinclude:: /examples/test_check_shapes_examples.py
+     :start-after: [argument_ref_index]
+     :end-before: [argument_ref_index]
+     :dedent:
+
+* Use ``[all]`` to select all elements of a collection:
+
+  .. literalinclude:: /examples/test_check_shapes_examples.py
+     :start-after: [argument_ref_all]
+     :end-before: [argument_ref_all]
+     :dedent:
 
-* Use ``.<name>`` to refer to attributes of the object.
-* Use ``[<index>]`` to refer to elements of a sequence. This is particularly useful if your
-  function returns a tuple of values.
-* Use ``[all]`` to select all elements in a collection.
-* Use ``.keys()`` to select all keys in a mapping.
-* Use ``.values()`` to select all values in a mapping.
+* Use ``.keys()`` to select all keys of a mapping:
 
-We do not support looking up values in a  ``dict``.
+  .. literalinclude:: /examples/test_check_shapes_examples.py
+     :start-after: [argument_ref_keys]
+     :end-before: [argument_ref_keys]
+     :dedent:
 
-If the argument, or any of the looked-up values, are `None` the check is skipped.
+* Use ``.values()`` to select all values of a mapping:
 
-For example::
+  .. literalinclude:: /examples/test_check_shapes_examples.py
+     :start-after: [argument_ref_values]
+     :end-before: [argument_ref_values]
+     :dedent:
 
-    @check_shapes(
-        "weights: ...",
-        "data.training_data: ...",
-        "return: ...",
-        "return[0]: ...",
-        "data_list[all]: ...",
-        "data_dict.keys(): ...",
-        "data_dict.values(): ...",
-        "something[all].foo[0].bar.values(): ...",
-    )
-    def f(...):
-        ...
+We do not support looking up a specific key or value in a  ``dict``.
```

Comment:
Can you make this a `.. note::`-block so that it pops up in the docs?

Reply:
> Can you make this a `.. note::`-block so that it pops up in the docs?

Sure.